src/reducers/real/index.js

```diff
@@ -0,0 +1,90 @@
+import {
+  ITEM_ADD,
+  ITEM_DELETE,
+  ITEM_RENAME,
+  CONTAINER_ADD,
+  CONTAINER_DELETE,
+  CONTAINER_RENAME,
+  SET_REAL_DATA
+} from 'actions/actiontypes'
+
+const initialState = {
+  _id: '',
+  name: '',
+  items: [],
+  modified_timestamp: '',
+  containers: [],
+  is_deleted: false,
+  timestamp: '',
+  snapshots: [],
+  unsnapshot_items: []
+}
+
+const realReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ITEM_ADD:
+      return {
+        ...state,
+        items: [
+          ...state.items,
+          action.item
+        ],
+        unsnapshot_items: [
+          ...state.unsnapshot_items,
+          action.item
+        ]
+      }
+    case ITEM_DELETE:
+      let items = state.items
+      items = items.filter(ele => ele._id !== action.id)
+      return {
+        ...state,
+        items: items
+      }
+    case ITEM_RENAME:
+      items = state.items
+      let item = items.find(ele => ele._id === action.item._id)
+      item.name = action.item.name
+      items = items.filter(ele => ele._id !== action.item._id)
+      return {
+        ...state,
+        items: [
+          ...state.items,
+          item
+        ]
+      }
+    case CONTAINER_ADD:
```

Comment:
same for containers lets, make a test for them as well.
And I think you will have to add the containers to all the snapshots.